### PR TITLE
Added a new section for Deployment and testing guidelines for Developers

### PR DESCRIPTION
**Pull request by VaniHaripriya**
## The issue resolved by this Pull Request:
Resolves https://github.com/opendatahub-io/data-science-pipelines-operator/issues/159

## Description of your changes:
Added new section with Deployment and testing process for developers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -467,3 +468,20 @@ They are as follows:
...
 [kfp-tekton]: https://github.com/kubeflow/kfp-tekton
+
+# Deployment and Testing Guidelines for Developers
+
+To run the DSPO locally :
+
+Clone the directory and execute the following command:
+
+```bash
+go run main.go --config=$config.yaml
```
Comment (DharmitD) [inline README.md:479]: Could you list out contents for the `config.yaml` file?

File: README.md [hunk 2]
```diff
@@ -467,3 +468,20 @@ They are as follows:
...
+```
+To run the tests:
+ 
+Execute `make test` or `make unittest` or `make functest` based on the level of testing that needs to be done.
+
+To deploy DSPO as a developer :
+
+Follow the instructions from [here](#deploy-the-operator-standalone) to deploy the operator standalone.
```
Comment (DharmitD) [inline README.md:487]: You could also mention deploying the operator via ODH/kfdef

**Review by DharmitD — COMMENTED** [review]
/lgtm

File: README.md [hunk 3]
```diff
@@ -467,3 +468,39 @@ They are as follows:
 [flipcoin code example]: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/docs/example_pipelines/condition.py
 [installodh]: https://opendatahub.io/docs/quick-installation
 [kfp-tekton]: https://github.com/kubeflow/kfp-tekton
+
+# Deployment and Testing Guidelines for Developers
```
[the remark this answers is not in the document]
Comment (VaniHaripriya) [reply README.md:472]: Thank you @HumairAK , will cover the suggested points.

File: README.md [hunk 4]
```diff
@@ -467,3 +468,39 @@ They are as follows:
...
+  MariaDB: registry.redhat.io/rhel8/mariadb-103:1-188
+  MlmdEnvoy: quay.io/opendatahub/ds-pipelines-metadata-envoy:1.7.0
+  MlmdGRPC: quay.io/opendatahub/ds-pipelines-metadata-grpc:1.0.0
+  MlmdWriter: quay.io/opendatahub/ds-pipelines-metadata-writer:1.1.0
+  CRDViewer: gcr.io/ml-pipeline/viewer-crd-controller:2.0.0-rc.2
+  VisualizationServer: gcr.io/ml-pipeline/visualization-server:2.0.0-rc.2
+```
+To run the tests:
+ 
+Execute `make test` or `make unittest` or `make functest` based on the level of testing that needs to be done.
```
Comment (gmfrasca) [inline README.md:500]: Let's expand on what each of these do.  example: functests are functional tests (what does that mean?), `make test` just runs both unittests and functests, etc

File: README.md [hunk 5]
```diff
@@ -467,3 +468,39 @@ They are as follows:
 [flipcoin code example]: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/docs/example_pipelines/condition.py
 [installodh]: https://opendatahub.io/docs/quick-installation
 [kfp-tekton]: https://github.com/kubeflow/kfp-tekton
+
+# Deployment and Testing Guidelines for Developers
+
+To run the DSPO locally :
```
Comment (gmfrasca) [inline README.md:474]: We should also mention that you need to be logged-into the OpenShift cluster via `oc login`, and CRDs need to be installed (how do we do that?)

File: README.md (hunk 3 above)
Comment (gmfrasca) [inline README.md:472]: We also should add subsections for:
- handling/running pre-commit
- building/generating manifests

File: README.md [hunk 6]
```diff
@@ -467,3 +468,39 @@ They are as follows:
...
+
+Clone the directory and execute the following command:
+
+```bash
+go run main.go --config=$config.yaml
+```
+Below is the sample config file(the tags for the images can be edited as required):
+```bash
+Images:
+  ApiServer: quay.io/opendatahub/ds-pipelines-api-server:main-0e8a011
```
Comment (gmfrasca) [inline README.md:484]: we should update these images to use latest tags (rather than explicit SHAs/dev tags) where applicable

File: README.md [hunk 7]
```diff
@@ -481,20 +494,27 @@ go run main.go --config=$config.yaml
...
   Cache: registry.access.redhat.com/ubi8/ubi-minimal
   MoveResultsImage: registry.access.redhat.com/ubi8/ubi-micro
   MariaDB: registry.redhat.io/rhel8/mariadb-103:1-188
-  MlmdEnvoy: quay.io/opendatahub/ds-pipelines-metadata-envoy:1.7.0
-  MlmdGRPC: quay.io/opendatahub/ds-pipelines-metadata-grpc:1.0.0
-  MlmdWriter: quay.io/opendatahub/ds-pipelines-metadata-writer:1.1.0
+  MlmdEnvoy: quay.io/opendatahub/ds-pipelines-metadata-envoy:latest
+  MlmdGRPC: quay.io/opendatahub/ds-pipelines-metadata-grpc:latest
+  MlmdWriter: quay.io/opendatahub/ds-pipelines-metadata-writer:latest
   CRDViewer: gcr.io/ml-pipeline/viewer-crd-controller:2.0.0-rc.2
```
Comment (gmfrasca) [inline README.md:508]: let's remove these 2 images, as I don't believe we'll be using introducing them now (from unrelated spikes/decisions during this sprint)

File: README.md [hunk 8]
```diff
@@ -504,3 +524,48 @@ To deploy DSPO as a developer :
...
+
+How to do disable health checks when dev testing:
+
+To disable the health checks set the values to true in the DSPA yaml file you apply. Refer to his sample file [here][dspa-yaml].
+
+In certain scenarios, it may be necessary to disable health checks within our environment. When the DSPO is executed either locally or on a different cluster, the health checks can't reach the database and Object Store endpoints. Consequently, they remain unsuccessful, preventing the deployment of essential pipeline infrastructure components by the DSPA. To address this challenge, we have introduced the `disableHealthCheck` mechanism as a viable solution.
+
+How to enable kfp ui and minio:
+
+Refer to this [sample][sample-yaml] yaml file to know how to enable kfp ui and minio.
```
Comment (gmfrasca) [inline README.md:569]:
```suggestion
Refer to this [sample][sample-yaml] yaml file for enabling the kfp ui and minio. 
```

File: README.md [hunk 9]
```diff
@@ -5,23 +5,45 @@ Data Science Pipeline stacks onto individual OCP namespaces.
...
+rerun these experiments.
+
+Data Science Pipelines is based on the upstream [Kubeflow Pipelines (KFP)][kfp]
+project. We leverage the [kfp-tekton][kfp] project to run pipelines backed
+by the Tekton (rather than Argo, which is the default choice in KFP). We
+currently distributed version 1.x of KFP, and are working to support v2.
+
+Data Scientists can use tools like the
+[kfp-tekton SDK](https://github.com/kubeflow/kfp-tekton/blob/master/sdk/README.md)
+or [Elyra](https://github.com/elyra-ai/elyra) to author their workflows, and
```
Comment (HumairAK) [inline README.md:44]: this commit should not show up, you'll need to rebase over this change

**Review by DharmitD — COMMENTED** [review]
/lgtm

**Comment by DharmitD** [comment]
/lgtm cancel

Looks like there are some merge conflicts and the PR needs rebase, could you please resolve them?

**Comment by VaniHaripriya** [comment]
Opening a new PR.

**Pull request closed without merging**